modules/Users/Login.php

```diff
@@ -174,63 +178,19 @@
 
     $captcha_privatekey = $admin->settings['captcha_private_key'];
     $captcha_publickey = $admin->settings['captcha_public_key'];
-    $captcha_js .=
-        "<script type='text/javascript' src='" .
-        getJSPath('cache/include/javascript/sugar_grp1_yui.js') . "'></script><script type='text/javascript' src='" .
-        getJSPath('cache/include/javascript/sugar_grp_yui2.js') . "'></script>
-			<script type='text/javascript' src='http://www.google.com/recaptcha/api/js/recaptcha_ajax.js'></script>
-			<script>
-			function initCaptcha(){
-			Recaptcha.create('$captcha_publickey' ,'captchaImage',{theme:'custom'});
-			}
-			window.onload=initCaptcha;
-
-			var handleFailure=handleSuccess;
-			var handleSuccess = function(o){
-				if(o.responseText!==undefined && o.responseText ==='Success'){
-					generatepwd();
-					Recaptcha.reload();
-				}
-				else{
-					if(o.responseText!=='')
-						document.getElementById('generate_success').innerHTML =o.responseText;
-					Recaptcha.reload();
-				}
-			}
-			var callback2 ={ success:handleSuccess, failure: handleFailure };
-
-			function validateAndSubmit(){
-					var form = document.getElementById('form');
-					var url = 
-					'&to_pdf=1&module=Home&action=index&entryPoint=Changenewpassword&recaptcha_challenge_field='+
-					Recaptcha.get_challenge()+'&recaptcha_response_field='+ Recaptcha.get_response();
-					YAHOO.util.Connect.asyncRequest('POST','index.php',callback2,url);
-			}</script>";
-    $Captcha .= "<tr>
-			<td scope='row' width='20%'>" . $mod_strings['LBL_RECAPTCHA_INSTRUCTION'] . ":</td>
-		    <td width='70%'><input type='text' size='26' id='recaptcha_response_field' value=''></td>
-
-		</tr>
-		<tr>
-
-		 	<td colspan='2'><div style='margin-left:2px'class='x-sqs-list' id='recaptcha_image'></div></td>
-		</tr>
-		<tr>
-			<td colspan='2' align='right'><a href='javascript:Recaptcha.reload()'>" .
-        $mod_strings['LBL_RECAPTCHA_NEW_CAPTCHA'] . "</a>&nbsp;&nbsp;
-			 		<a class='recaptcha_only_if_image' href='javascript:Recaptcha.switch_type(\"audio\")'>" .
-        $mod_strings['LBL_RECAPTCHA_SOUND'] . "</a>
-			 		<a class='recaptcha_only_if_audio' href='javascript:Recaptcha.switch_type(\"image\")'> " .
-        $mod_strings['LBL_RECAPTCHA_IMAGE'] . '</a>
-		 	</td>
-		</tr>';
-    $sugar_smarty->assign('CAPTCHA', $Captcha);
-    echo $captcha_js;
+    $captchaContentTemplate = new Sugar_Smarty();
+    $captchaContentTemplate->assign('APP_LIST_STRINGS', $app_list_strings);
```

Comment:
since we added these language strings to Sugar_Smarty object, it doesn't necessary.
https://github.com/salesagility/SuiteCRM/blob/hotfix/include/Sugar_Smarty.php#L146

Reply:
I have added a util file to remedy the issue (include/utils/recaptcha_utils.php).